app/services/ipam/base.rb

```diff
@@ -17,6 +17,18 @@ def initialize(opts = {})
       normalize_mac!
     end
 
+    def subnet_range
+      @subnet_range ||= begin
+        subnet_range = IPAddr.new("#{subnet.network}/#{subnet.mask}", subnet.family).to_range
+        # exclude first element - network
+        from = subnet.from.present? ? IPAddr.new(subnet.from) : subnet_range.first(2).last
+        # exclude last element - broadcast
```

Comment:
I think it is safe to subtract two IP addresses from whole subnet when range is not present, it is needed for IPv4 and won't hurt for IPv6. But I can do this only in IPv4 case if you insist.